platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/FelixConfig.java

```diff
@@ -0,0 +1,114 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for enabling functional constructs with {@link Configuration}s and exposes the
+ * felix.fileinstall.filename property, if it exists, as a {@link File}.
+ */
+public class FelixConfig {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FelixConfig.class);
+
+    private static final String FELIX_FILENAME_PROP = "felix.fileinstall.filename";
+
+    private final Configuration config;
+
+    private final File felixFile;
+
+    FelixConfig(Configuration config) {
+        if (config == null) {
+            throw new IllegalArgumentException("Configuration can't be null");
+        }
+        this.config = config;
+        this.felixFile = extractFelixFileInstallProp(config);
+    }
+
+    /**
+     * @return the pid for this configuration.
+     */
+    String getPid() {
+        return config.getPid();
+    }
+
+    /**
+     * @return the File object pointing to the this configuration on disk.
+     */
+    @Nullable
+    File getFile() {
+        return felixFile;
+    }
+
+    /**
+     * Determine if the felix.fileinstall.filename property changed.
+     *
+     * @param original file location to compare to the current file location poitned to by this
+     *                 config.
+     * @return False if the file locations by absolute path are identical. True otherwise.
+     */
+    boolean filePropChanged(File original) {
+        return !Objects.equals(felixFile, original);
+    }
+
+    private File extractFelixFileInstallProp(Configuration config) {
+        if (config.getProperties() == null) {
+            LOGGER.debug("Config properties are null, they may have just been created");
+            return null;
+        }
+        Object felixConfigFileName = config.getProperties()
+                .get(FELIX_FILENAME_PROP);
+        if (felixConfigFileName == null) {
+            LOGGER.debug("Felix filename prop not found for pid {}", config.getPid());
+            return null;
+        }
+        return createFileFromFelixProp(felixConfigFileName);
+    }
+
+    /**
+     * Code adopted from Karaf's Config Repository Impl:
+     * https://github.com/apache/karaf/blob/master/config/src/main/java/org/apache/karaf/config/core/impl/
+     * ConfigRepositoryImpl.java#L101-L114
+     */
+    private File createFileFromFelixProp(Object felixConfigFileName) {
+        try {
+            if (felixConfigFileName instanceof URL) {
+                return new File(((URL) felixConfigFileName).toURI());
+            }
+            if (felixConfigFileName instanceof URI) {
+                return new File((URI) felixConfigFileName);
+            }
+            if (felixConfigFileName instanceof String) {
+                return new File(new URL((String) felixConfigFileName).toURI());
+            }
+        } catch (URISyntaxException | MalformedURLException e) {
+            LOGGER.debug("Was expecting a correctly formatted URL or URI, but got: ", e);
```

Comment:
✏️ Should log the `felixConfigFileName` that failed to assist in debugging. Unless the two exceptions will always include the failed name.